### ISSUE TITLE
Help with vanilla Qemu + Netmap

### ISSUE THREAD

**Issue by leleobhz** [luigirizzo/netmap#367]
Hello!

I've built Qemu 2.10 from fedora git (Using fedora patches and vanilla qemu) and I'm facing a issue I can figure out what is happening.

I compiled it adding --enable-netmap in the configure script and Ive set my libvirt as following:

```
<domain type='kvm' xmlns:qemu='http://libvirt.org/schemas/domain/qemu/1.0'>
...
  </devices>
  <qemu:commandline>
    <qemu:arg value='-netdev'/>
    <qemu:arg value='type=netmap,ifname=netmap:pipe0{1,id=netmap0'/>
    <qemu:arg value='-device'/>
    <qemu:arg value='virtio-net-pci,netdev=netmap0,mac=52:54:00:58:00:52,ioeventfd=on,mrg_rxbuf=on'/>
  </qemu:commandline>
</domain>                    
```

When I start the machine, the folliowing error happens:

```
[root@deodora ~]# virsh start netmap-experiments-01
erro:Falha ao iniciar o domínio netmap-experiments-01
erro:internal error: process exited while connecting to monitor: 2017-09-14T13:25:52.396002Z qemu-system-x86_64: -chardev pty,id=charserial0: char device redirected to /dev/pts/6 (label charserial0)
552.400266 nm_open [920] cannot open /dev/netmap: No such file or directory pipe0{1
2017-09-14T13:25:52.400285Z qemu-system-x86_64: -netdev type=netmap,ifname=netmap:pipe0{1,id=netmap0: Failed to nm_open() netmap:pipe0{1: No such file or directory

[root@deodora ~]# LANG=C virsh start netmap-experiments-01
error: Failed to start domain netmap-experiments-01
error: internal error: process exited while connecting to monitor: 2017-09-14T13:29:52.361597Z qemu-system-x86_64: -chardev pty,id=charserial0: char device redirected to /dev/pts/6 (label charserial0)
792.365759 nm_open [920] cannot open /dev/netmap: No such file or directory pipe0{1
2017-09-14T13:29:52.365781Z qemu-system-x86_64: -netdev type=netmap,ifname=netmap:pipe0{1,id=netmap0: Failed to nm_open() netmap:pipe0{1: No such file or directory
```

If I test python support, things happens ok:

```
In [1]: import netmap

In [2]: test0=netmap.NetmapDesc('netmap:pipe0{1')

In [3]: test1=netmap.NetmapDesc('netmap:pipe0}1')

In [4]: print test0.interface()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-cd0374624eb7> in <module>()
----> 1 print test0.interface()

TypeError: 'netmap.NetmapInterface' object is not callable

In [5]: print test0.interface
name: 'pipe0{1'
version:    0
flags:      0x1
tx_rings:   1
rx_rings:   1
bufs_head:  0
spare1[0]:  0x0
spare1[1]:  0x0
spare1[2]:  0x0
spare1[3]:  0x0
spare1[4]:  0x0


In [6]: print test1.interface
name: 'pipe0}1'
version:    0
flags:      0x1
tx_rings:   1
rx_rings:   1
bufs_head:  0
spare1[0]:  0x0
spare1[1]:  0x0
spare1[2]:  0x0
spare1[3]:  0x0
spare1[4]:  0x0
```

So, why Qemu cannot open the interface? I think QEmu may be instancing netmap correctly because the error quotes the /dev/netmap.

Also, even running it as root, I've changed the permission of the file:

```
[root@deodora ~]# ls -lah /dev/netmap
crw-rw-rw-. 1 root root 10, 58 set 13 16:13 /dev/netmap
```

And I think there is no SELinux violation:

```
[root@deodora ~]# cat /var/log/audit/audit.log | audit2allow 


#============= cockpit_session_t ==============
allow cockpit_session_t init_t:dbus send_msg;

#============= groupadd_t ==============
allow groupadd_t init_t:dbus send_msg;

#============= init_t ==============
allow init_t rpm_script_t:dbus send_msg;

#============= local_login_t ==============
allow local_login_t init_t:dbus send_msg;

#============= sshd_t ==============
allow sshd_t init_t:dbus send_msg;

#============= syslogd_t ==============
allow syslogd_t self:netlink_generic_socket create;

#============= useradd_t ==============
allow useradd_t init_t:dbus send_msg;

#============= virtlogd_t ==============
allow virtlogd_t system_dbusd_t:dbus send_msg;
[root@deodora ~]# 
```

What can I do?

Thanks!
```

**Comment by leleobhz** [comment]
Just one more information:

```
[root@deodora ~]# strings /usr/bin/qemu-system-x86_64 | grep -i net_init_netmap
net_init_netmap
[root@deodora ~]# qemu-system-x86_64 --version
QEMU emulator version 2.10.0(qemu-2.10.0-1.fc26)
Copyright (c) 2003-2017 Fabrice Bellard and the QEMU Project developers
```

**Comment by vmaffione** [comment]
Your QEMU gets ENOENT on `open()` here https://github.com/luigirizzo/netmap/blob/master/sys/net/netmap_user.h#L808. QEMU calls `nm_open()` here https://github.com/qemu/qemu/blob/master/net/netmap.c#L95.

If open() fails with ENOENT, it means that `linux_netmap_open()` is not even called, because netmap kernel module does not return ENOENT on open(). You can confirm it by putting a `D("called");` here https://github.com/luigirizzo/netmap/blob/master/LINUX/netmap_linux.c#L1110).

In conclusion, this means there is some misconfiguration in your machine. Maybe libvirt runs within some non-root cgroups or namespaces, which explains why it does not see /dev/netmap.

**Comment by vmaffione** [comment]
Closing, please reopen if needed
